From 63dc545f3ed9dc6c8c1c7a4ff274a1a04a341b1b Mon Sep 17 00:00:00 2001
From: Rui Loura <rui@joyent.com>
Date: Fri, 1 Feb 2019 18:58:37 +0000
Subject: [PATCH] TRITON-1163 cnapi should use agentIpFromSysinfo() for
 mockcloud support Reviewed by: Josh Wilsdon <josh@wilsdon.ca> Approved by:
 Josh Wilsdon <josh@wilsdon.ca>

---
 lib/models/server.js | 15 ++++++++-------
 package.json         |  2 +-
 2 files changed, 9 insertions(+), 8 deletions(-)

diff --git a/lib/models/server.js b/lib/models/server.js
index 8ac9892..0d399a4 100644
--- a/lib/models/server.js
+++ b/lib/models/server.js
@@ -1884,13 +1884,14 @@ function (opts) {
             });
         },
         function getCnapiLocationFromSysinfo(wfcb) {
-            if (sysinfo.hasOwnProperty('CN Agent IP')) {
-                // Allow sysinfo to include an IP that we'll use to connect to
-                // that might be different from the "Admin IP".
-                serverAdminIp = sysinfo['CN Agent IP'];
-            } else {
-                serverAdminIp = netconfig.adminIpFromSysinfo(sysinfo);
-            }
+            /*
+             * agentIpFromSysinfo() is intended for use with mockcloud which
+             * requires the use of a different IP for contacting agents on a
+             * mock CN.  If mockcloud support is not a requirement,
+             * adminIpFromSysinfo() should be used to get the admin IP for a
+             * CN.
+             */
+            serverAdminIp = netconfig.agentIpFromSysinfo(sysinfo);
             if (!serverAdminIp) {
                 wfcb(new VError('parsing server ip address in sendTaskReq '
                     + '(No admin NICs detected.)'));
diff --git a/package.json b/package.json
index 5cc1d16..667323a 100644
--- a/package.json
+++ b/package.json
@@ -26,7 +26,7 @@
     "sprintf": "0.1.5",
     "trace-event": "1.3.0",
     "triton-metrics": "1.0.0",
-    "triton-netconfig": "1.1.0",
+    "triton-netconfig": "1.3.0",
     "vasync": "1.6.4",
     "verror": "1.9.0",
     "wf-client": "0.2.1",
-- 
2.21.0

